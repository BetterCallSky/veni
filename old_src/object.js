import _makeError from './_makeError';
import _optional from './_optional';
import pipe from './pipe';

export default (msg = 'must be an') => pipe(
  _optional(),
  (data) => {
    if (typeof data.value !== 'object') {
      return {
        stop: true,
        error: _makeError(data, msg),
      };
    }
    return undefined;
  },
);
