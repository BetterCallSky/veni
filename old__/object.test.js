import object from '../src/object';

describe('object', () => {
  [undefined, null].forEach(value => {
    it(`should ignore optional: ${value}`, () => {
      expect(object()({value})).toMatchSnapshot();
    });
  });

  [1, 'abc', [1, 2, 3]].forEach(value => {
    it(`should return an error for: ${value}`, () => {
      expect(object()({value})).toMatchSnapshot();
    });
  });

  it('should return a custom message', () => {
    expect(object('custom message')({value: 12})).toMatchSnapshot();
  });

  it('should return a value for a valid object', () => {
    expect(object()({value: {foo: 'bar'}})).toMatchSnapshot();
  });
});
