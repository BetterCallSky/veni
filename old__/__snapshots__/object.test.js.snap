// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object should ignore optional: null 1`] = `
Object {
  "errors": Array [],
  "value": null,
}
`;

exports[`object should ignore optional: undefined 1`] = `
Object {
  "errors": Array [],
  "value": undefined,
}
`;

exports[`object should return a custom message 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "custom message",
      "path": "value",
    },
  ],
  "value": 12,
}
`;

exports[`object should return a value for a valid object 1`] = `
Object {
  "errors": Array [],
  "value": Object {
    "foo": "bar",
  },
}
`;

exports[`object should return an error for: 1 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "must be an",
      "path": "value",
    },
  ],
  "value": 1,
}
`;

exports[`object should return an error for: 1,2,3 1`] = `
Object {
  "errors": Array [],
  "value": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`object should return an error for: abc 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "must be an",
      "path": "value",
    },
  ],
  "value": "abc",
}
`;
